pkg/phlare/modules.go

```diff
@@ -104,14 +103,63 @@ func (f *Phlare) initQueryFrontend() (services.Service, error) {
 	return frontendSvc, nil
 }
 
-type fakeLimits struct{}
+func (f *Phlare) initRuntimeConfig() (services.Service, error) {
+	if len(f.Cfg.RuntimeConfig.LoadPath) == 0 {
+		// no need to initialize module if load path is empty
+		return nil, nil
+	}
+
+	f.Cfg.RuntimeConfig.Loader = loadRuntimeConfig
+
+	// make sure to set default limits before we start loading configuration into memory
+	validation.SetDefaultLimitsForYAMLUnmarshalling(f.Cfg.LimitsConfig)
+
+	serv, err := runtimeconfig.New(f.Cfg.RuntimeConfig, prometheus.WrapRegistererWithPrefix("phlare_", f.reg), log.With(f.logger, "component", "runtime-config"))
+	if err == nil {
+		// TenantLimits just delegates to RuntimeConfig and doesn't have any state or need to do
+		// anything in the start/stopping phase. Thus we can create it as part of runtime config
+		// setup without any service instance of its own.
+		f.TenantLimits = newTenantLimits(serv)
+	}
+
+	f.RuntimeConfig = serv
+
+	f.Server.HTTP.Methods("GET").Path("/runtime_config").Handler(runtimeConfigHandler(f.RuntimeConfig, f.Cfg.LimitsConfig))
+	f.Server.HTTP.Methods("GET").Path("/api/v1/user_limits").Handler(middleware.AuthenticateUser.Wrap(validation.UserLimitsHandler(f.Cfg.LimitsConfig, f.TenantLimits)))
```

Comment:
I really want to get rid of `user`  when we actually mean `tenant`.  I think now is the time

```suggestion
	f.Server.HTTP.Methods("GET").Path("/api/v1/tenant_limits").Handler(middleware.AuthenticateUser.Wrap(validation.UserLimitsHandler(f.Cfg.LimitsConfig, f.TenantLimits)))
```